src/app/api/webhook/webhook.service.ts

```diff
@@ -120,53 +140,82 @@ class WebhookService extends BaseService {
       previousAttributes: { companyId: prevCompanyId },
     } = data
     // If company hasn't been changed - don't bother with any of this
-    if (prevCompanyId === newCompanyId) return
+    if (prevCompanyId === newCompanyId || !prevCompanyId) return
 
     // Only delete prev notifications if client had a valid company before
-    const copilot = new CopilotAPI(this.user.token)
-    const prevCompany = prevCompanyId ? await copilot.getCompany(prevCompanyId) : null
+    let prevCompany, newCompany
+    try {
+      prevCompany = await this.copilot.getCompany(prevCompanyId)
+    } catch (e: unknown) {
+      prevCompany = null
+    }
+    try {
+      newCompany = await this.copilot.getCompany(newCompanyId)
+    } catch (e: unknown) {
+      newCompany = null
+    }
+
+    if (!prevCompany?.name && newCompany?.name) {
```

Comment:
Can a company name be empty? 
Just asking because if prev company if is falsy, we are returning above.

Reply:
Yes, in case of individual clients a unique company is generated for them.
This company returns a result in their retrieve company endpoint, and the name is an empty string.

The above snippet checks for both falsy prevCompany as well as empty string in prevCompany.name.